src/battle_ai_main.c

```diff
@@ -3536,15 +3564,19 @@ static s32 AI_CheckViability(u32 battlerAtk, u32 battlerDef, u32 move, s32 score
     case EFFECT_HAZE:
         if (AnyStatIsRaised(BATTLE_PARTNER(battlerAtk))
           || PartnerHasSameMoveEffectWithoutTarget(BATTLE_PARTNER(battlerAtk), move, aiData->partnerMove))
+        {
             ADJUST_SCORE(-3);
             break;
-        // fallthrough
+        }
+        score += AI_TryToClearStats(battlerAtk, battlerDef, isDoubleBattle);
+        break;
     case EFFECT_ROAR:
-    case EFFECT_CLEAR_SMOG:
-        if (isDoubleBattle)
-            score += min(CountPositiveStatStages(battlerDef) + CountPositiveStatStages(BATTLE_PARTNER(battlerDef)), 7);
-        else
-            score += min(CountPositiveStatStages(battlerDef), 4);
+        if ((gBattleMoves[move].soundMove && aiData->abilities[battlerDef] == ABILITY_SOUNDPROOF) || aiData->abilities[battlerDef] == ABILITY_SUCTION_CUPS)
+        {
+            ADJUST_SCORE(-3);
+            break;
+        }
+        score += AI_TryToClearStats(battlerAtk, battlerDef, FALSE);
```

Comment:
Something isn't adding up here. It looks like you are handling clear smog in effect roar.

Reply:
No?